pkg/proxy/proxy.go

```diff
@@ -142,6 +149,181 @@ func (p *Proxy) allocatePort() (uint16, error) {
+		if c.Labels != nil {
+			info.Labels = c.Labels.Labels.GetModel()
+			info.LabelsSHA256 = c.Labels.GetLabelsSHA256()
+		}
+	}
+}
+
+// getInfoFromConsumable fills the accesslog.EndpointInfo fields, by fetching
+// the consumable from the consumable cache of endpoint using identity sent by
+// source. This is needed in ingress proxy while logging the source endpoint
+// info.  Since there will be 2 proxies on the same host, if both egress and
+// ingress policies are set, the ingress policy cannot determine the source
+// endpoint info based on ip address, as the ip address would be that of the
+// egress proxy i.e host.
+func getInfoFromConsumable(ipstr string, info *accesslog.EndpointInfo,
+	srcIdentity policy.NumericIdentity, r Redirect) {
+	ep := r.getSource()
+	ip := net.ParseIP(ipstr)
+	if ip != nil {
+		if ip.To4() != nil {
+			info.IPv4 = ip.String()
+		} else {
+			info.IPv6 = ip.String()
+		}
+	}
+	secIdentity := ep.ResolveIdentity(srcIdentity)
+
+	if secIdentity != nil {
+		info.Labels = secIdentity.Labels.GetModel()
+		info.LabelsSHA256 = secIdentity.Labels.SHA256Sum()
+		info.Identity = uint64(srcIdentity)
+	}
+}
+
+// egressDestinationInfo returns the destination EndpointInfo for a flow
+// leaving the proxy at egress.
+func egressDestinationInfo(ipstr string, info *accesslog.EndpointInfo) {
+	ip := net.ParseIP(ipstr)
+	if ip != nil {
+		if ip.To4() != nil {
+			info.IPv4 = ip.String()
+
+			if nodeaddress.IsHostIPv4(ip) {
+				fillReservedIdentity(info, policy.ReservedIdentityHost)
+				return
+			}
+
+			if nodeaddress.GetIPv4ClusterRange().Contains(ip) {
+				c := addressing.DeriveCiliumIPv4(ip)
+				ep := endpointmanager.LookupIPv4(c.String())
+				if ep != nil {
+					ep.RLock()
+					info.ID = uint64(ep.ID)
+					info.Labels = ep.GetLabels()
+					info.LabelsSHA256 = ep.GetLabelsSHA()
+					info.Identity = uint64(ep.GetIdentity())
+					ep.RUnlock()
+				} else {
+					fillReservedIdentity(info, policy.ReservedIdentityCluster)
+				}
+			} else {
+				fillReservedIdentity(info, policy.ReservedIdentityWorld)
+			}
+		} else {
+			info.IPv6 = ip.String()
+
+			if nodeaddress.IsHostIPv6(ip) {
+				fillReservedIdentity(info, policy.ReservedIdentityHost)
+				return
+			}
+
+			if nodeaddress.GetIPv6ClusterRange().Contains(ip) {
+				c := addressing.DeriveCiliumIPv6(ip)
+				id := c.EndpointID()
+				info.ID = uint64(id)
+
+				ep := endpointmanager.LookupCiliumID(id)
+				if ep != nil {
+					ep.RLock()
+					info.Labels = ep.GetLabels()
+					info.LabelsSHA256 = ep.GetLabelsSHA()
+					info.Identity = uint64(ep.GetIdentity())
+					ep.RUnlock()
+				} else {
+					fillReservedIdentity(info, policy.ReservedIdentityCluster)
+				}
+			} else {
+				fillReservedIdentity(info, policy.ReservedIdentityWorld)
+			}
+		}
+	}
+}
+
+// GetDestinationInfo returns the destination EndpointInfo.
+func GetDestinationInfo(dstIPPort string, r Redirect, ingress bool) accesslog.EndpointInfo {
```

Comment:
This can still be made private

Reply:
sure.. my concepts on private functions in a golang package are a bit shaky :) Will do

Reply:
done